src/net_processing.cpp

```diff
@@ -2187,6 +2251,67 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
     }
 
 
+    else if (strCommand == NetMsgType::TX_DANDELION) {
+        // Stop processing the transaction early if
+        // dandelion is disabled and peer is either not whitelisted or whitelistrelay is off
+        if (!CNode::IsDandelionEnabled()&& (!pfrom->fWhitelisted || !gArgs.GetBoolArg("-whitelistrelay", DEFAULT_WHITELISTRELAY))) {
```

Comment:
Some IRC discussion around not extending whitelist-relay behavior further with dandelion processing:
https://botbot.me/freenode/bitcoin-core-dev/2018-08-13/?msg=103271568&page=3